### PR TITLE
Create find_tests() and execute_tasks() functions + some refactoring.

### PR DESCRIPTION
**Pull request by ehlemur-zz**

### CODE REVIEW

**Comment by ehlemur-zz** [comment]
@pbos @kwiberg 
Please take a look :)

**Comment by pbos** [comment]
@kwiberg would you mind taking this one? :)

File: gtest-parallel [hunk 1]
```diff
@@ -336,6 +334,94 @@ class TestTimes(object):
...
+    def __call__(self):
+      while True:
+        task = None
+        with self.task_lock:
+          if self.task_id < len(self.tasks):
+            task = self.tasks[self.task_id]
+          self.task_id += 1
+        if task is None:
+          return
+        task.run()
```
Comment (kwiberg) [inline gtest-parallel]: I think you can write this as something like
```
  while True:
    with self.task_lock:
      if self.task_id < len(self.tasks):
        task = self.tasks[self.task_id]
        self.task_id += 1
      else:
        return
    task.run()
```

File: gtest-parallel [hunk 2]
```diff
@@ -336,6 +334,94 @@ class TestTimes(object):
...
+    t = threading.Thread(target=func)
+    t.daemon = True
+    t.start()
+    return t
+
+  try:
+    timeout.start()
+    worker_fn = WorkerFn(tasks)
+    workers = [start_daemon(worker_fn) for _ in range(options.workers)]
+    [t.join() for t in workers]
```
Comment (kwiberg) [inline gtest-parallel]: Less compactly, but more conventionally, written as
```
for t in workers:
  t.join()
```
I know you've only moved this line, though, so feel free to not change it.

**Review by kwiberg — APPROVED** [review]
Would you like to make changes, or merge as is?

**Comment by ehlemur-zz** [comment]
I'll make those changes.

**Comment by ehlemur-zz** [comment]
Don't merge yet, please

File: gtest-parallel [hunk 3]
```diff
@@ -336,6 +334,95 @@ class TestTimes(object):
...
+def execute_tasks(tasks):
+  class WorkerFn(object):
+    def __init__(self, tasks):
+      self.task_id = 0
+      self.tasks = tasks
+      self.task_lock = threading.Lock()
+
+    def __call__(self):
+      while True:
+        task = None
```
Comment (kwiberg) [inline gtest-parallel]: This line isn't necessary anymore, is it?

**Review by kwiberg — CHANGES_REQUESTED** [review]

File: gtest-parallel (hunk 3 above)
Comment (ehlemur-zz) [reply gtest-parallel]: Right.

**Comment by ehlemur-zz** [comment]
Please take a look

**Review by kwiberg — APPROVED** [review]

**Pull request merged**